### PR TITLE
Adding multiweight support to SSD

### PR DESCRIPTION
**Pull request by datumbox**
Fixes partially #4675

Verified that both commands return the same:
```
torchrun --nproc_per_node=1 train.py --test-only --pretrained --model ssd300_vgg16

torchrun --nproc_per_node=1 train.py --test-only --weights Coco_RefV1 --model ssd300_vgg16
```

cc @datumbox @bjuncek

### CODE REVIEW

File: torchvision/prototype/models/detection/ssd.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    trainable_backbone_layers: Optional[int] = None,
+    **kwargs: Any,
+) -> SSD:
+    if "pretrained" in kwargs:
+        warnings.warn("The argument pretrained is deprecated, please use weights instead.")
+        weights = SSD300VGG16Weights.Coco_RefV1 if kwargs.pop("pretrained") else None
+    weights = SSD300VGG16Weights.verify(weights)
+    if "pretrained_backbone" in kwargs:
+        warnings.warn("The argument pretrained_backbone is deprecated, please use weights_backbone instead.")
+        weights_backbone = VGG16Weights.ImageNet1K_Features if kwargs.pop("pretrained_backbone") else None
```
Comment (datumbox) [inline torchvision/prototype/models/detection/ssd.py:52]: We are using special weights here. Not the standard VGG weights of TorchVision.

File: torchvision/prototype/models/detection/ssd.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        scales=[0.07, 0.15, 0.33, 0.51, 0.69, 0.87, 1.05],
+        steps=[8, 16, 32, 64, 100, 300],
+    )
+
+    defaults = {
+        # Rescale the input in a way compatible to the backbone
+        "image_mean": [0.48235, 0.45882, 0.40784],
+        "image_std": [1.0 / 255.0, 1.0 / 255.0, 1.0 / 255.0],  # undo the 0-1 scaling of toTensor
+    }
+    kwargs: Any = {**defaults, **kwargs}
```
Comment (datumbox) [inline torchvision/prototype/models/detection/ssd.py:80]: Any type declaration needed to mypy.

File: torchvision/prototype/models/vgg.py [hunk 3]
```diff
@@ -106,6 +106,23 @@ class VGG16Weights(Weights):
             "acc@5": 90.382,
         },
     )
+    # We port the features of a VGG16 backbone trained by amdegroot because unlike the one on TorchVision, it uses the
+    # same input standardization method as the paper. Only the `features` weights have proper values, those on the
+    # `classifier` module are filled with nans.
+    ImageNet1K_Features = WeightEntry(
```
Comment (datumbox) [inline torchvision/prototype/models/vgg.py:112]: We are adding the weights as a separate entry on the VGG. This one only has the features part of the model.

File: torchvision/prototype/models/vgg.py [hunk 4]
```diff
@@ -106,6 +106,23 @@ class VGG16Weights(Weights):
...
+        transforms=partial(
+            ImageNetEval, crop_size=224, mean=(0.48235, 0.45882, 0.40784), std=(1.0 / 255.0, 1.0 / 255.0, 1.0 / 255.0)
+        ),
+        meta={
+            "size": (224, 224),
+            "categories": None,
+            "interpolation": InterpolationMode.BILINEAR,
+            "recipe": "https://github.com/amdegroot/ssd.pytorch#training-ssd",
+            "acc@1": float("nan"),
+            "acc@5": float("nan"),
```
Comment (datumbox) [inline torchvision/prototype/models/vgg.py:123]: It doesn't contain a classifier, hence the nans here to denote it.

**Review by datumbox — COMMENTED** [review]
Adding few comments to assist review:

File: torchvision/prototype/models/detection/ssd.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    model = SSD(backbone, anchor_generator, (300, 300), num_classes, **kwargs)
```
Comment (jdsgomes) [inline torchvision/prototype/models/detection/ssd.py:81]: should we get the size from the weights?

**Review by jdsgomes — COMMENTED** [review]
LGTM, just left a minor comment for consideration

File: torchvision/prototype/models/detection/ssd.py (hunk 5 above)
Comment (datumbox) [reply torchvision/prototype/models/detection/ssd.py:81]: It's the same discussion as in https://github.com/pytorch/vision/pull/4875#r744621284.

Note that unlike the FasterRCNN models that adapt better on variable input size, SSD doesn't. It was investigated at #3819 and that's why it's hardcoded.

**Review by jdsgomes — APPROVED** [review]
Thanks

**Pull request merged**